### ISSUE TITLE
[Bug] when run with read-only access, chainsaw returns a zero exitcode

### ISSUE THREAD

**Issue by lnattrass** [kyverno/chainsaw#2001]
### chainsaw version Version

v0.2.10

### Description

When running chainsaw with not enough/invalid privileges, it errors out with a "FAIL" but still returns an exit code of 0.

### Steps to reproduce

1. With this test
```
apiVersion: chainsaw.kyverno.io/v1alpha1
kind: Test
metadata:
  name: example
spec:
  steps:
  - name: snooze
    try:
    - sleep:
        duration: 30s
```

2. Execute chainsaw as a `read-only` user (or even a `default` service account) against a namespace that doesn't yet exist, for example:

```
~/work/chainsaw-err $ chainsaw test --namespace kokoman; echo $?
Version: 0.2.10
Loading default configuration...
- Using test file: chainsaw-test
- TestDirs [.]
- SkipDelete false
- FailFast false
- Namespace 'kokoman'
- FullName false
- IncludeTestRegex ''
- ExcludeTestRegex ''
- ApplyTimeout 5s
- AssertTimeout 30s
- CleanupTimeout 30s
- DeleteTimeout 15s
- ErrorTimeout 30s
- ExecTimeout 5s
- DeletionPropagationPolicy Background
- Template true
- NoCluster false
- PauseOnFailure false
Loading tests...
- example (.)
Loading values...
Running tests...
=== RUN   chainsaw
=== PAUSE chainsaw
=== CONT  chainsaw
    | 15:21:53 | chainsaw | @chainsaw | CREATE    | WARN  | v1/Namespace @ kokoman
        === ERROR
        namespaces is forbidden: User "read-only/lnattrass" cannot create resource "namespaces" in API group "" at the cluster scope
    | 15:21:53 | chainsaw | @chainsaw | INTERNAL  | ERROR |
        === ERROR
        namespaces is forbidden: User "read-only/lnattrass" cannot create resource "namespaces" in API group "" at the cluster scope
--- FAIL: chainsaw (1.35s)
FAIL
Tests Summary...
- Passed  tests 0
- Failed  tests 0
- Skipped tests 0
Done.
0
```

### Expected behavior

If the tests failed, it should exit > 0

Our test infrastructure failed to catch that we hadn't provided enough privileges to one of our tests, because it's exiting 0.

### Troubleshooting

- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by eddycharly** [comment]
Good catch, thanks for reporting!
Fixing...

**Comment by eddycharly** [comment]
Closing, fixed in #2002

**Comment by lnattrass** [comment]
Thank you for the fast turnaround!
